ryo_ninomiya step5

ryo-ninomiya opened the pull request:
# 課題 (STEP5)
[STEP5](https://github.com/Fablic/training/blob/develop/steps_jp.md#user-content-ステップ5-作りたいアプリケーションのイメージを考える)
- 設計を進める前に、どのようなアプリになるか完成イメージを考える。
- システムの要件を読んで、必要なデータ構造を考える。
- データ構造を考えたら、それをモデル図に手書きで書いてみる。

# やったこと
- アプリイメージの作成
- テーブルスキームの作成
- [Readme](https://github.com/Fablic/training/blob/ryo_ninomiya-step5/myapp/README.md) にアプリイメージとテーブルスキームをドキュメント化


レビューをお願いします。

Commit:
step5 readmeにDB schemaを追加

Commit:
半角スペースを追加

ryo-ninomiya commented:
PR作成前の@u-Hashimotoさんからのレビューバック

## DB設計
ドキュメント化のおすすめ
この内容をREADME上か何か別ドキュメントかに記載しておくと良いと思います！
研修ステップの中でテーブル変更が出てきた時に都度更新するイメージです。（自動でドキュメントを更新するような仕組みを作る強者もいるそうです...)
ドキュメント化する際は各種制約（pk,fk,unique,index)と論理名も欲しいです 👀

## 設計の粒度
結構最後の方のSTEPまで見て設計されているのかなと思っています。
最初からきっちり決めていただいても良いですし、start時点では必要最低限のカラムだけ設計、でも良いかもしれません。
（実際の開発では必要に応じて仕様や設計が追加・変更されるものなので、各STEPに入ったときに必要なものを追加する、実開発の練習みたいな感じです。）

登録日/更新日
Railsの御作法で、登録日・更新日はcreated_at、updated_atにすると良いです。
（migration実行するときに指定してないと自動的にこのカラムが作られます。）特にこだわりがなければ揃えた方が良さそうです！

## Tasksテーブル
beadline
deadlineですかね？
urgency、importance
これは画面のデザインには見当たりませんが、後々出てくる感じでしょうか。個人的にはLabelで充分そうだなと思いました。
こだわりがあれば持っていても良いと思います！

## Labelsテーブル
リレーションについて
このシステム上、ラベルは誰かが作ったら誰でも使えるもの、という設計でしょうか。
userとの紐付けは不要ですか？（絶対いるというわけではなくて、仕様によると思うので確認です！）
color
intということはカラーコードではなくてenum的なもので管理の予定ですかね。

u-Hashimoto reviewed (commented):
確認しました！
 - create_at ,update_at
create**d**_at,update**d**_at でお願いします 🙏 

 -  Users
 email:verchar(45)で足りますか？ 👀 

 - belong_tasks
おそらくテーブル名はtasks_labelsとかになるかなと思います。
ですが、このテーブルは初回からは作らないと思うので（ラベル実装段階で作ると思います）、後々の修正でもいいかもです。
関連テーブルを考えるステップでRailsの御作法的にどういう名前をつけようかなと考えることになると思いますので。

FYI:
これから先、テーブル名や関数名・変数名などに迷った場合、Railsの公式ガイドを参考にすると色々ヒントが載ってて良いかと思いますです！
https://railsguides.jp/association_basics.html

Commit:
step5 レビューバック修正
created_at,updated_atに修正
email:verchar(45) -> (255)
belong_tasks -> tasks_labels

ryo-ninomiya commented:
@u-Hashimoto
レビューありがとうございます！
頂いた指摘箇所の修正を行いましたので再度レビューをお願いします。

u-Hashimoto reviewed (approved):
いいと思います！
![LGTM](https://image.lgtmoon.dev/2457)
